<h1> ACTIVITY 1 : PALMER PENGUIN DATA ANALYTICS</h1>
<hr>
<h3> Analyst : Quert Russel Lalisan </h3>

In [1]:
#import

import numpy as np
import pandas as pd

<h1> Converting Dataset(s) into DataFrame </h1>

<h2> Requirements </h2>
<li> Name of the Dataframe</li>
<li> File name</li>

In [2]:
penguins = pd.read_csv('dataset\\penguins.csv')

In [3]:
penguins

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen            39.1           18.7              181.0   
1       Adelie  Torgersen            39.5           17.4              186.0   
2       Adelie  Torgersen            40.3           18.0              195.0   
3       Adelie  Torgersen             NaN            NaN                NaN   
4       Adelie  Torgersen            36.7           19.3              193.0   
..         ...        ...             ...            ...                ...   
339  Chinstrap      Dream            55.8           19.8              207.0   
340  Chinstrap      Dream            43.5           18.1              202.0   
341  Chinstrap      Dream            49.6           18.2              193.0   
342  Chinstrap      Dream            50.8           19.0              210.0   
343  Chinstrap      Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
0         3750.0    male  2007  
1         3800.0  female  2007  
2         3250.0  female  2007  
3            NaN     NaN  2007  
4         3450.0  female  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[344 rows x 8 columns]

<h2> Summerization </h2>

In [4]:
penguins.isnull().sum()

species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

In [5]:
penguins.describe()

bill_length_mm  bill_depth_mm  flipper_length_mm  body_mass_g  \
count      342.000000     342.000000         342.000000   342.000000   
mean        43.921930      17.151170         200.915205  4201.754386   
std          5.459584       1.974793          14.061714   801.954536   
min         32.100000      13.100000         172.000000  2700.000000   
25%         39.225000      15.600000         190.000000  3550.000000   
50%         44.450000      17.300000         197.000000  4050.000000   
75%         48.500000      18.700000         213.000000  4750.000000   
max         59.600000      21.500000         231.000000  6300.000000   

              year  
count   344.000000  
mean   2008.029070  
std       0.818356  
min    2007.000000  
25%    2007.000000  
50%    2008.000000  
75%    2009.000000  
max    2009.000000

<h2> Filtering </h2>

In [6]:
penguins['species'].value_counts()

species
Adelie       152
Gentoo       124
Chinstrap     68
Name: count, dtype: int64

In [7]:
penguins[penguins['island'] == 'Dream']

species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
30      Adelie  Dream            39.5           16.7              178.0   
31      Adelie  Dream            37.2           18.1              178.0   
32      Adelie  Dream            39.5           17.8              188.0   
33      Adelie  Dream            40.9           18.9              184.0   
34      Adelie  Dream            36.4           17.0              195.0   
..         ...    ...             ...            ...                ...   
339  Chinstrap  Dream            55.8           19.8              207.0   
340  Chinstrap  Dream            43.5           18.1              202.0   
341  Chinstrap  Dream            49.6           18.2              193.0   
342  Chinstrap  Dream            50.8           19.0              210.0   
343  Chinstrap  Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
30        3250.0  female  2007  
31        3900.0    male  2007  
32        3300.0  female  2007  
33        3900.0    male  2007  
34        3325.0  female  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[124 rows x 8 columns]

In [8]:
penguins[penguins['sex'] == 'male']

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen            39.1           18.7              181.0   
5       Adelie  Torgersen            39.3           20.6              190.0   
7       Adelie  Torgersen            39.2           19.6              195.0   
13      Adelie  Torgersen            38.6           21.2              191.0   
14      Adelie  Torgersen            34.6           21.1              198.0   
..         ...        ...             ...            ...                ...   
334  Chinstrap      Dream            50.2           18.8              202.0   
336  Chinstrap      Dream            51.9           19.5              206.0   
339  Chinstrap      Dream            55.8           19.8              207.0   
341  Chinstrap      Dream            49.6           18.2              193.0   
342  Chinstrap      Dream            50.8           19.0              210.0   

     body_mass_g   sex  year  
0         3750.0  male  2007  
5         3650.0  male  2007  
7         4675.0  male  2007  
13        3800.0  male  2007  
14        4400.0  male  2007  
..           ...   ...   ...  
334       3800.0  male  2009  
336       3950.0  male  2009  
339       4000.0  male  2009  
341       3775.0  male  2009  
342       4100.0  male  2009  

[168 rows x 8 columns]

In [9]:
penguins[penguins['bill_depth_mm'] >= 20]

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
5       Adelie  Torgersen            39.3           20.6              190.0   
9       Adelie  Torgersen            42.0           20.2              190.0   
13      Adelie  Torgersen            38.6           21.2              191.0   
14      Adelie  Torgersen            34.6           21.1              198.0   
17      Adelie  Torgersen            42.5           20.7              197.0   
19      Adelie  Torgersen            46.0           21.5              194.0   
35      Adelie      Dream            39.2           21.1              196.0   
36      Adelie      Dream            38.8           20.0              190.0   
49      Adelie      Dream            42.3           21.2              191.0   
61      Adelie     Biscoe            41.3           21.1              195.0   
85      Adelie      Dream            41.3           20.3              194.0   
101     Adelie     Biscoe            41.0           20.0              203.0   
103     Adelie     Biscoe            37.8           20.0              190.0   
107     Adelie     Biscoe            38.2           20.0              190.0   
111     Adelie     Biscoe            45.6           20.3              191.0   
114     Adelie     Biscoe            39.6           20.7              191.0   
117     Adelie  Torgersen            37.3           20.5              199.0   
137     Adelie      Dream            40.2           20.1              200.0   
287  Chinstrap      Dream            51.7           20.3              194.0   
292  Chinstrap      Dream            50.3           20.0              197.0   
305  Chinstrap      Dream            52.8           20.0              205.0   
307  Chinstrap      Dream            54.2           20.8              201.0   
313  Chinstrap      Dream            52.0           20.7              210.0   

     body_mass_g     sex  year  
5         3650.0    male  2007  
9         4250.0     NaN  2007  
13        3800.0    male  2007  
14        4400.0    male  2007  
17        4500.0    male  2007  
19        4200.0    male  2007  
35        4150.0    male  2007  
36        3950.0    male  2007  
49        4150.0    male  2007  
61        4400.0    male  2008  
85        3550.0    male  2008  
101       4725.0    male  2009  
103       4250.0    male  2009  
107       3900.0    male  2009  
111       4600.0    male  2009  
114       3900.0  female  2009  
117       3775.0    male  2009  
137       3975.0    male  2009  
287       3775.0    male  2007  
292       3300.0    male  2007  
305       4550.0    male  2008  
307       4300.0    male  2008  
313       4800.0    male  2008

In [10]:
penguins[penguins['year'] == 2005]

Empty DataFrame
Columns: [species, island, bill_length_mm, bill_depth_mm, flipper_length_mm, body_mass_g, sex, year]
Index: []

In [11]:
pingwing = penguins[(penguins['species'] == 'Adelie') & (penguins['bill_length_mm'] >= 30)]
pingwing

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0    Adelie  Torgersen            39.1           18.7              181.0   
1    Adelie  Torgersen            39.5           17.4              186.0   
2    Adelie  Torgersen            40.3           18.0              195.0   
4    Adelie  Torgersen            36.7           19.3              193.0   
5    Adelie  Torgersen            39.3           20.6              190.0   
..      ...        ...             ...            ...                ...   
147  Adelie      Dream            36.6           18.4              184.0   
148  Adelie      Dream            36.0           17.8              195.0   
149  Adelie      Dream            37.8           18.1              193.0   
150  Adelie      Dream            36.0           17.1              187.0   
151  Adelie      Dream            41.5           18.5              201.0   

     body_mass_g     sex  year  
0         3750.0    male  2007  
1         3800.0  female  2007  
2         3250.0  female  2007  
4         3450.0  female  2007  
5         3650.0    male  2007  
..           ...     ...   ...  
147       3475.0  female  2009  
148       3450.0  female  2009  
149       3750.0    male  2009  
150       3700.0  female  2009  
151       4000.0    male  2009  

[151 rows x 8 columns]

<h2> Getting Infos </h2>

In [12]:
pingwing.info()

<class 'pandas.core.frame.DataFrame'>
Index: 151 entries, 0 to 151
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   species            151 non-null    object 
 1   island             151 non-null    object 
 2   bill_length_mm     151 non-null    float64
 3   bill_depth_mm      151 non-null    float64
 4   flipper_length_mm  151 non-null    float64
 5   body_mass_g        151 non-null    float64
 6   sex                146 non-null    object 
 7   year               151 non-null    int64  
dtypes: float64(4), int64(1), object(3)
memory usage: 10.6+ KB


In [13]:
penguins['bill_length_mm'].mean()

np.float64(43.9219298245614)

In [14]:
penguins.isnull().sum()

species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

<h2> Changing The Values </h2>

In [15]:
penguins['bill_length_mm'] = penguins['bill_length_mm'].fillna(penguins['bill_length_mm'].mean())
penguins['bill_depth_mm'] = penguins['bill_depth_mm'].fillna(penguins['bill_depth_mm'].mean())
penguins['flipper_length_mm'] = penguins['flipper_length_mm'].fillna(penguins['flipper_length_mm'].mean())
penguins['body_mass_g'] = penguins['body_mass_g'].fillna(penguins['body_mass_g'].mean())
penguins['sex'] = penguins['sex'].fillna(penguins['sex'].mode()[0])
penguins.isnull().sum()

species              0
island               0
bill_length_mm       0
bill_depth_mm        0
flipper_length_mm    0
body_mass_g          0
sex                  0
year                 0
dtype: int64

<h2> Changing Category </h2>

In [16]:
penguins['species'] = penguins['species'].astype('category')
penguins.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype   
---  ------             --------------  -----   
 0   species            344 non-null    category
 1   island             344 non-null    object  
 2   bill_length_mm     344 non-null    float64 
 3   bill_depth_mm      344 non-null    float64 
 4   flipper_length_mm  344 non-null    float64 
 5   body_mass_g        344 non-null    float64 
 6   sex                344 non-null    object  
 7   year               344 non-null    int64   
dtypes: category(1), float64(4), int64(1), object(2)
memory usage: 19.4+ KB


In [17]:
penguins['mass_kg'] = np.round(penguins['body_mass_g']/1000,1)
penguins

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen        39.10000       18.70000         181.000000   
1       Adelie  Torgersen        39.50000       17.40000         186.000000   
2       Adelie  Torgersen        40.30000       18.00000         195.000000   
3       Adelie  Torgersen        43.92193       17.15117         200.915205   
4       Adelie  Torgersen        36.70000       19.30000         193.000000   
..         ...        ...             ...            ...                ...   
339  Chinstrap      Dream        55.80000       19.80000         207.000000   
340  Chinstrap      Dream        43.50000       18.10000         202.000000   
341  Chinstrap      Dream        49.60000       18.20000         193.000000   
342  Chinstrap      Dream        50.80000       19.00000         210.000000   
343  Chinstrap      Dream        50.20000       18.70000         198.000000   

     body_mass_g     sex  year  mass_kg  
0    3750.000000    male  2007      3.8  
1    3800.000000  female  2007      3.8  
2    3250.000000  female  2007      3.2  
3    4201.754386    male  2007      4.2  
4    3450.000000  female  2007      3.4  
..           ...     ...   ...      ...  
339  4000.000000    male  2009      4.0  
340  3400.000000  female  2009      3.4  
341  3775.000000    male  2009      3.8  
342  4100.000000    male  2009      4.1  
343  3775.000000  female  2009      3.8  

[344 rows x 9 columns]

<h1> 1) Which species of penguins does the island of Dream have the most? </h1>

In [18]:
penguins[penguins["island"] == 'Dream']['species'].describe()

count           124
unique            2
top       Chinstrap
freq             68
Name: species, dtype: object

<h3> Insigths : The species that has the most amount in the island of dream is Chinstrap. The chinstrap having a populaton of 68 out of the 124 occupatant in the island. </h3>

<h1> 2) Which species has the most population</h1>

In [19]:
penguins['species'].describe()

count        344
unique         3
top       Adelie
freq         152
Name: species, dtype: object

<h3> Insights : Adelie is the most plentiful species of penguin with a population of 152</h3>

<h1>3) What is the male to female ratio in the island of Torgersen </h1>

In [20]:
penguins[penguins['island'] == 'Torgersen']['sex'].describe()


count       52
unique       2
top       male
freq        28
Name: sex, dtype: object

<h3> Insights : The male to female is 7:6 where there are 7 male for every 6 females in the island of Torgersen </h3>

 <h1>4) What is the average bill length for a female Gentoo in the year 2007 </h1>

In [21]:
np.round(penguins[
    (penguins['species'] == 'Gentoo') &
    (penguins['sex'] == 'female') &
    (penguins['year'] == 2007)
]['bill_length_mm'].mean(0))



np.float64(45.0)

<h3> Insights : The average bill length for a female Gentoo in the year 2007 is 45 </h3>

<h1>5) What is the heaviest recorded penguin on Torgersen Island? </h1> 

In [22]:
penguins[penguins['island']=='Torgersen']['mass_kg'].max()
penguins["species"].value_counts()

(penguins[
        (penguins['island'] =='Torgersen') & 
        (penguins['mass_kg'] == 4.7)
 ])

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
7   Adelie  Torgersen            39.2           19.6              195.0   
81  Adelie  Torgersen            42.9           17.6              196.0   

    body_mass_g   sex  year  mass_kg  
7        4675.0  male  2007      4.7  
81       4700.0  male  2008      4.7

<h3> Insights : There are two of the heaviest recorded penguin on Torgesen Island, both are a species of Adelie with a body mass kilogram of "4.7" </h3>

<h1>6) Do male Gentoo penguins weigh more than females on average? </h1>

In [23]:
penguins[
    penguins['species'] == 'Gentoo']['sex'].describe()

count      124
unique       2
top       male
freq        66
Name: sex, dtype: object

In [24]:
male = np.round(penguins[
    (penguins['species'] == 'Gentoo') &
    (penguins['sex']=='male') 
]['mass_kg']).mean()

female = np.round(penguins[
    (penguins['species'] == 'Gentoo') &
    (penguins['sex']=='female') 
]['mass_kg']).mean()

print(f"Female has '{female}' in average, while male has '{male}'")

Female has '4.724137931034483' in average, while male has '5.46969696969697'


<h3>Insights: Yes, the species Gentoo has a population of 124 of compromising of mostly male with 66, and having the average weight of 5.47 kg while female penguins has an average weight of 4.72, therefore male Gentoo penguins weigh more in average more than females.</h3>

<h1>7) Which species has the longest average bill length?</h1>

In [75]:
penguins['species'].describe()

count        344
unique         3
top       Adelie
freq         152
Name: species, dtype: object

In [70]:
Gentoo_ave = np.round(penguins[penguins['species'] == 'Gentoo']['bill_length_mm'].mean())
Adelie_ave = np.round(penguins[penguins['species'] == 'Adelie']['bill_length_mm'].mean())
Chinstrap_ave = np.round(penguins[penguins['species'] == 'Chinstrap']['bill_length_mm'].mean())

print(f'Gento has : {Gentoo_ave} mm\nAdelie has : {Adelie_ave} mm\nChinstrap has {Chinstrap_ave} mm')

Gento has : 47.0 mm
Adelie has : 39.0 mm
Chinstrap has 49.0 mm


<h3>Insights : Out of all the penguin species <b>'Chinstrap'</b> has the longest average bill length, Averaging up to <b>49 mm</b> outmatching Gentoo and the Adelie species  </h3>